kotlin/show-location-history/src/main/java/com/esri/arcgisruntime/sample/showlocationhistory/MainActivity.kt

```diff
@@ -102,8 +102,14 @@ class MainActivity : AppCompatActivity() {
       initialZoomScale = 7000.0
     }
 
-    // change the isTrackLocation flag and the button's icon
+    // reset location display and change the isTrackLocation flag and the button's icon
     button.setOnClickListener {
+      // if the user has panned away from the location display, turn it on again
+      if (mapView.locationDisplay.autoPanMode == LocationDisplay.AutoPanMode.OFF) {
+        mapView.locationDisplay.autoPanMode = LocationDisplay.AutoPanMode.RECENTER
+        // don't change the isTrackLocation status if the button is recentering
+        return@setOnClickListener
```

Comment:
I actually wonder if this is actually more confusing?

It's more intuitive to me that the toggle always toggles on tap and will re-center if applicable. So, I'd suggest dropping the return. Otherwise, looks good!

Reply:
Yeah, that makes sense to me. The only other option would be to have two buttons, which I don't think is necessary here